src/visualization/types/Graph/view.tsx

```diff
@@ -243,7 +248,12 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
       }
     }
     // everything is under the 'default' category for now:
-    const selectedAnnotations: any[] = annotations?.default ?? []
+    let selectedAnnotations: any[] = []
+
+    visibleAnnotationStreams.map(stream => {
+      selectedAnnotations = selectedAnnotations.concat(annotations[stream.id])
+    })
```

Comment:
`map` will return a new array with each item described by the function, you shouldn't perform mutative operations in them (like modifying a variable that is external to the function).

[`filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) will return an array of items that pass the test described by the function, or an empty array. I think that's what you want here. Something like:

```js
const selectedAnnotations: any[] = visibleAnnotationStreams.filter(stream => {
  if (annotations[stream.id]) {
    return annotations[stream.id]
  }
})